src/oge/helpers.py

```diff
@@ -296,21 +393,33 @@ def add_plant_nameplate_capacity(df: pd.DataFrame) -> pd.DataFrame:
     Returns:
         pd.DataFrame: original data frame with additional 'capacity_mw' column.
     """
-    generators_capacity = load_data.load_pudl_table(
+    generator_capacity = load_data.load_pudl_table(
         "generators_eia860",
+        year=earliest_data_year,
+        end_year=latest_validated_year,
         columns=["plant_id_eia", "generator_id", "report_date", "capacity_mw"],
+    ).sort_values(by=["plant_id_eia", "generator_id", "report_date"], ascending=True)
+
+    generator_capacity["capacity_mw"] = generator_capacity.groupby(
+        ["plant_id_eia", "generator_id"]
+    )["capacity_mw"].bfill()
+    generator_capacity["capacity_mw"] = generator_capacity.groupby(
+        ["plant_id_eia", "generator_id"]
+    )["capacity_mw"].ffill()
+
+    # keep only the most recent year of data
+    generator_capacity = generator_capacity.drop_duplicates(
```

Comment:
In the case of nameplate capacity, I think that we only want to keep the specific data year, not the latest validated year. Nameplate capacity can chance over time if the generator is repowered, so this value might be annually varying.

Reply:
It's still good that we load all years and do the fill in case there is missing capacity data in a specific year.